### ISSUE TITLE
[MTB] User impersonation does not work

### ISSUE THREAD

**Issue by JimBugwadia** [kubernetes-sigs/multi-tenancy#942]
I have a namespace test, with a role binding defined:

```
➜  multi-tenancy git:(master) ✗ kubectl -n test describe rolebinding
Name:         jane-role
Labels:       <none>
Annotations:  <none>
Role:
  Kind:  ClusterRole
  Name:  edit
Subjects:
  Kind  Name  Namespace
  ----  ----  ---------
  User  jane
```
➜  multi-tenancy git:(master) ✗ kubectl -n test get rolebinding jane-role -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  creationTimestamp: "2020-07-22T21:40:09Z"
  name: jane-role
  namespace: test
  resourceVersion: "1854717"
  selfLink: /apis/rbac.authorization.k8s.io/v1/namespaces/test/rolebindings/jane-role
  uid: 52c4db73-41a0-4e48-b8f8-95815aacb4c4
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: edit
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: User
  name: jane
```
 
I can use this via kubectl:

```
➜  multi-tenancy git:(master) ✗ kubectl get ns --as jane
Error from server (Forbidden): namespaces is forbidden: User "jane" cannot list resource "namespaces" in API group "" at the cluster scope
➜  multi-tenancy git:(master) ✗ kubectl -n kube-system get pods --as jane
Error from server (Forbidden): pods is forbidden: User "jane" cannot list resource "pods" in API group "" in the namespace "kube-system"
➜  multi-tenancy git:(master) ✗ kubectl -n test get pods --as jane
No resources found in test namespace.
```

However, I am unable to run the benchmarks as this user:

```
➜  multi-tenancy git:(master) ✗ kubectl-mtb test benchmarks -n test --as jane
error: Make sure you have entered valid tenant-admin and tenant namespace.
```

**Comment by phoenixking25** [comment]
@JimBugwadia i also faced a serviceaccount. So we have created a check in which it performs like 
`kubectl auth can-i get ns/<tenantnamepsace> --as Serviceaccount ` plus with a user too
So unless the tenantnamepsace is created by a user/SA, it will not be accessible by any other user/SA. If you create tenantnamespace with a user and then run benchmark only then it will run. I dont know if this a valid check or not, but we applied this check so that besides tenantadmin(who has created tenantnamespaces) no one can else run benchmarks.
Another reason we had to validate the user that is given by the user as without this check some of benchmarks were getting passed if you pass any random string as user.
